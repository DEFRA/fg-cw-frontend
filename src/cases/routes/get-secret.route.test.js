import { load } from "cheerio";
import { afterAll, beforeAll, describe, expect, it, vi } from "vitest";
import { createServer } from "../../server/index.js";
import { findSecretUseCase } from "../use-cases/find-secret.use-case.js";
import { getSecretRoute } from "./get-secret.route.js";

vi.mock("../use-cases/find-secret.use-case.js");

describe("getSecret", () => {
  let server;

  beforeAll(async () => {
    server = await createServer();
    server.route(getSecretRoute);

    await server.initialize();
  });

  afterAll(async () => {
    await server.stop();
  });

  it("returns 200", async () => {
    const { statusCode, result } = await server.inject({
      method: "GET",
      url: "/secret",
      auth: {
        isAuthenticated: true,
        isAuthorized: false,
        isInjected: true,
        credentials: {
          token: "access-token",
          refreshToken: "refresh-token",
          expiresAt: new Date("2050-01-01T00:00:00Z").getTime(),
          user: {
            id: "43e8508b6cbd4ac1b29ee73792ab0f4b",
            name: "Joe Bloggs",
            email: "joe@bloggs.com",
            idpRoles: ["FCP.Casework.Read"],
            appRoles: ["ROLE_SING_AND_DANCE"],
          },
        },
        strategy: "session",
        mode: "required",
        error: null,
      },
    });

    expect(statusCode).toEqual(200);

    const $ = load(result);
    const view = $("#main-content").html();

    expect(findSecretUseCase).toHaveBeenCalledWith("access-token");

    expect(view).toMatchSnapshot();
  });
});
