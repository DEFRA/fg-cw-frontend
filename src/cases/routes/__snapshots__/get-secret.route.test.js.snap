// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`getSecret > returns 200 1`] = `
"
  <div class="govuk-grid-row app-heading" data-testid="app-heading">
  <div class="govuk-grid-column-two-thirds">
    <h1 class="govuk-heading-xl govuk-!-margin-bottom-2" data-testid="app-heading-title">
      
    </h1>

  </div>
</div>


  <div class="govuk-grid-row">
    <div class="govuk-grid-column-full">
      <div class="govuk-body" data-testid="app-page-body">
        <h2>Shhh - it's a secret!</h2>
        <p>
          You are
            authenticated
        </p>
        <p>
          You are
authorised        </p>

          <a href="/logout" role="button" draggable="false" class="govuk-button" data-module="govuk-button">
  Logout
</a>

        <h3>Session information</h3>
        <pre>{
  "isAuthenticated": true,
  "isAuthorized": false,
  "isInjected": true,
  "credentials": {
    "profile": {
      "id": "43e8508b-6cbd-4ac1-b29e-e73792ab0f4b",
      "displayName": "Joe Bloggs",
      "email": "joe@bloggs.com"
    },
    "authenticated": true,
    "authorised": true
  },
  "artifacts": {
    "profile": {
      "id": "43e8508b-6cbd-4ac1-b29e-e73792ab0f4b",
      "displayName": "Joe Bloggs",
      "email": "joe@bloggs.com"
    },
    "authenticated": true,
    "authorised": true
  },
  "strategy": "session",
  "mode": "required",
  "error": null
}</pre>
      </div>
    </div>
  </div>
        "
`;
